### ISSUE TITLE
Looks like the key will be leaked in the request

### ISSUE THREAD

**Issue by saviourdog** [ghuyfel/flutter_azure_tts#18]

**Comment by mikron123** [comment]
Yes, I don't understand why this package exists. Key should be on server side.

**Comment by GruppeGAM** [comment]
It does it leak or not?

**Comment by ghuyfel** [comment]
1. When making _HTTPS requests_, the headers and the body are encrypted in transit, and thus "can not" be leaked, unless you have them in clear in some form in your source code.
 
2. The only requests made with the subscription key are the ones to the `issueToken ` endpoint. In other words, I exchange the subscription key with a token and use that token to call the TTS endpoint. That token is valid for 10 minutes. When it expires I make another call to the `issueToken` endpoint to get a new one.

You can get more details [here](https://learn.microsoft.com/en-us/azure/ai-services/speech-service/rest-text-to-speech?tabs=streaming#authentication)

I hope this helps

**Comment by ghuyfel** [comment]
> It does it leak or not?

Please read my comment above
